tests/Aspire.Cli.Tests/Commands/PublishCommandTests.cs

```diff
@@ -22,4 +25,236 @@ public async Task PublishCommandWithHelpArgumentReturnsZero()
+        {
+            options.DotNetCliRunnerFactory = (sp) =>
+            {
+                var runner = new TestDotNetCliRunner();
+                runner.GetAppHostInformationAsyncCallback = (projectFile, cancellationToken) =>
+                {
+                    return (0, false, "9.0.0"); // Simulate an incompatible app host
+                };
+                return runner;
+            };
+        });
+
+        var provider = services.BuildServiceProvider();
+        var command = provider.GetRequiredService<RootCommand>();
+
+        // Act
+        var result = command.Parse("publish --project valid.csproj");
+        var exitCode = await result.InvokeAsync().WaitAsync(CliTestConstants.DefaultTimeout);
+
+        // Assert
+        Assert.NotEqual(0, exitCode); // Ensure the command fails
+    }
+
+    [Fact]
+    public async Task PublishCommandFailsWhenAppHostBuildFails()
+    {
+        // Arrange
+        var services = CliTestHelper.CreateServiceCollection(outputHelper, options =>
+        {
+            options.DotNetCliRunnerFactory = (sp) =>
+            {
+                var runner = new TestDotNetCliRunner();
+                runner.BuildAsyncCallback = (projectFile, cancellationToken) =>
+                {
+                    return 1; // Simulate a build failure
+                };
+                return runner;
+            };
+        });
+
+        var provider = services.BuildServiceProvider();
+        var command = provider.GetRequiredService<RootCommand>();
+
+        // Act
+        var result = command.Parse("publish --project valid.csproj");
+        var exitCode = await result.InvokeAsync().WaitAsync(CliTestConstants.DefaultTimeout);
+
+        // Assert
+        Assert.NotEqual(0, exitCode); // Ensure the command fails
+    }
+
+    [Fact]
+    public async Task PublishCommandFailsWhenAppHostCrashesBeforeBackchannelEstablished()
+    {
+        // Arrange
+        var services = CliTestHelper.CreateServiceCollection(outputHelper, options =>
+        {
+            options.DotNetCliRunnerFactory = (sp) =>
+            {
+                var runner = new TestDotNetCliRunner();
+
+                // Simulate a successful build
+                runner.BuildAsyncCallback = (projectFile, cancellationToken) => 0;
+
+                // Simulate apphost starting but crashing before backchannel is established
+                runner.RunAsyncCallback = async (projectFile, watch, noBuild, args, env, backchannelCompletionSource, cancellationToken) =>
+                {
+                    // Simulate a delay to mimic apphost starting
+                    await Task.Delay(100, cancellationToken);
+
+                    // Simulate apphost crash by completing the backchannel with an exception
+                    backchannelCompletionSource?.SetException(new InvalidOperationException("AppHost process has exited unexpectedly. Use --debug to see more deails."));
```

Comment:
```suggestion
                    backchannelCompletionSource?.SetException(new InvalidOperationException("AppHost process has exited unexpectedly. Use --debug to see more details."));
```

Reply:
Was actually a typo in the actual CLI as well. Fixed.